wporg-gp-translation-events.php

```diff
@@ -153,38 +153,35 @@ function validate_event_dates( $event_start, $event_end ) {
 }
 
 function submit_event_ajax() {
-	$event_id;
-	if ( 'create_event' === $_POST['form_name'] ) {
-		if ( ! isset( $_POST['create_event_nonce'] ) || ! wp_verify_nonce( $_POST['create_event_nonce'], 'create_event_nonce' ) ) {
-			wp_send_json_error( 'Nonce verification failed' );
-		}
-	}
-	if ( 'edit_event' === $_POST['form_name'] ) {
-		if ( ! isset( $_POST['edit_event_nonce'] ) || ! wp_verify_nonce( $_POST['edit_event_nonce'], 'edit_event_nonce' ) ) {
-			wp_send_json_error( 'Nonce verification failed' );
-		}
+	$event_id     = null;
+	$form_actions = array( 'draft', 'publish' );
+	if ( ! isset( $_POST['_event_nonce'] ) || ! wp_verify_nonce( $_POST['_event_nonce'], '_event_nonce' ) ) {
+		wp_send_json_error( 'Nonce verification failed' );
 	}
-
-	$title          = sanitize_text_field( $_POST['event_title'] );
-	$description    = sanitize_text_field( $_POST['event_description'] );
-	$event_start    = sanitize_text_field( $_POST['event_start'] );
-	$event_end      = sanitize_text_field( $_POST['event_end'] );
-	$locale         = sanitize_text_field( $_POST['event_locale'] );
-	$project_name   = sanitize_text_field( $_POST['event_project_name'] );
-	$event_timezone = sanitize_text_field( $_POST['event_timezone'] );
+	$title          = isset( $_POST['event_title'] ) ? sanitize_text_field( $_POST['event_title'] ) : '';
+	$description    = isset( $_POST['event_description'] ) ? sanitize_text_field( $_POST['event_description'] ) : '';
+	$event_start    = isset( $_POST['event_start'] ) ? sanitize_text_field( $_POST['event_start'] ) : '';
+	$event_end      = isset( $_POST['event_end'] ) ? sanitize_text_field( $_POST['event_end'] ) : '';
+	$locale         = isset( $_POST['event_locale'] ) ? sanitize_text_field( $_POST['event_locale'] ) : '';
+	$project_name   = isset( $_POST['event_project_name'] ) ? sanitize_text_field( $_POST['event_project_name'] ) : '';
+	$event_timezone = isset( $_POST['event_timezone'] ) ? sanitize_text_field( $_POST['event_timezone'] ) : '';
```

Comment:
If we eventually introduce an `Event` class (like suggested in another comment), this could be moved to a "static constructor" on that class, something like this:


```php
class Event {
	private DateTime $start;
	// ...

	private function __construct(
		DateTime $start,
		// ...
	) {
		$this->start = $start;
		// ...
	}

	public static function fromForm(): Event {
		$start = new DateTime(isset( $_POST['event_start'] ) ? sanitize_text_field( $_POST['event_start'] ) : '');

		return new Event(
			$start,
			// ...
		);
	}
}
```